torch/csrc/autograd/variable.h

```diff
@@ -428,19 +426,41 @@ inline Variable make_variable(at::Tensor data, Edge gradient_edge) {
 // Tensor Conversion
 //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-inline Variable& as_variable_ref(at::Tensor& tensor) {
-#ifdef DEBUG
+inline bool is_variable(const at::Tensor& tensor) noexcept {
   // dynamic_cast will return a nullptr if the `TensorImpl`'s dynamic type is
   // not `Variable::Impl`.
-  if (dynamic_cast<Variable::Impl*>(tensor.get()) == nullptr) {
-    throw std::runtime_error(
+  return dynamic_cast<const Variable::Impl*>(tensor.get()) != nullptr;
+}
+
+/// Downcasts the `Tensor` reference to a `Variable` reference. If compiling
+/// in DEBUG mode and the tensor's dynamic type is not in fact `Variable`,
+/// throws a `std::invalid_argument` exception.
+inline Variable& as_variable_ref(at::Tensor& tensor, bool check = false) {
```

Comment:
What do you think about pulling in the part of Ed's change (#5123) that adds `bool is_variable` to Type just always checking that `tensor` is truly a variable?